src/main/java/com/facebook/openwifirrm/optimizers/channel/LeastUsedChannelOptimizer.java

```diff
@@ -373,7 +376,8 @@ public Map<String, Map<String, Integer>> computeChannelMap() {
 				}
 
 				// Get current channel of the device
-				State state = model.latestState.get(serialNumber);
+				List<State> states = model.latestStates.get(serialNumber);
+				State state = states.get(states.size() - 1);
```

Comment:
You created a `latestState` earlier in this method - you could use that and replace these two lines with: `State state = latestState.get(serialNumber)`